### PR TITLE
Fix _scrollOffsetIsInBounds analayzer warning

### PR DESCRIPTION
**Pull request by HansMuller**

### CODE REVIEW

File: sky/packages/sky/lib/src/fn3/scrollable.dart [hunk 1]
```diff
@@ -126,7 +126,10 @@ abstract class ScrollableState<T extends Scrollable> extends State<T> {
   }
 
   bool _scrollOffsetIsInBounds(double offset) {
-    return offset >= scrollBehavior.minScrollOffset && offset < scrollBehavior.maxScrollOffset;
+    if (scrollBehavior is! ExtentScrollBehavior)
+      return false;
+    ExtentScrollBehavior behavior = scrollBehavior as ExtentScrollBehavior;
```
Comment (Hixie) [inline sky/packages/sky/lib/src/fn3/scrollable.dart]: don't need the as (and having it is more expensive in release mode). Just do the assignment.

**Comment by Hixie** [comment]
When would it be reasonable for scrollBehavior to not be an ExtentScrollBehavior?

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
though you should have a test to exercise this codepath.

File: sky/packages/sky/lib/src/fn3/scrollable.dart (hunk 1 above)
Comment (HansMuller) [inline sky/packages/sky/lib/src/fn3/scrollable.dart]: Good point, I've removed it.

**Comment by HansMuller** [comment]
I think I could concoct a ScrollBehavior that wasn't a ExtentScrollBehavior but AFIK all our own uses are extents based.

We need loads more Scrollable, ScrollBehavior and features and subclasses tests. I will add some.

**Pull request merged**